river/XwaylandOverrideRedirect.zig

```diff
@@ -114,7 +114,9 @@ pub fn handleMap(listener: *wl.Listener(void)) void {
 
 fn mapImpl(self: *Self) error{OutOfMemory}!void {
     const surface = self.xwayland_surface.surface.?;
-    self.surface_tree = try server.root.layers.xwayland_override_redirect.createSceneSubsurfaceTree(surface);
+    if (server.root.layers.xwayland_override_redirect) |xor| {
+        self.surface_tree = try xor.createSceneSubsurfaceTree(surface);
+    }
```

Comment:
There's no need for an `if` here, just assert that the scene tree is non-null with `.?`